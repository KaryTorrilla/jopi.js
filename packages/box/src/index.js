import styled from 'styled-components'
import {
  space,
  color,
  layout,
  flexbox,
  border,
  typography,
  position,
  variant,
} from 'styled-system'
import css from '@styled-system/css'

const sx = props => css(props.sx)(props.theme)
const base = props => css(props.__css)(props.theme)
const theme = ({ tx = 'variants' }) =>
  variant({
    scale: tx,
  })

export const Box = styled.div(
  {
    boxSizing: 'border-box',
    margin: 0,
    minWidth: 0,
  },
  base,
  theme,
  sx,
  space,
  color,
  layout,
  flexbox,
  typography,
  border,
  position
)

export const Flex = styled(Box)({
  display: 'flex',
})
