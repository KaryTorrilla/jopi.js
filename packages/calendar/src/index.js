import React from 'react'
import FullCalendar from '@fullcalendar/react' // must go before plugins
import dayGridPlugin from '@fullcalendar/daygrid' // a plugin!
import '@oneloop/theme/styles/globals.css'
import '../styles/styles.css'

export const Calendar = () => {
  /*   const calendarOptions = {
      events: [],
      plugins: [timeGridPlugin],
      initialView: 'timeGridDay',
      nowIndicator: true,
      headerToolbar: {
        start: 'prev',
        center: 'title',
        end: 'next',
      },
      editable: true,
      eventResizableFromStart: true,
      dragScroll: true,
      dayMaxEvents: 2,
      navLinks: true,
    } */

  return (
    <div className='calendarContainer'>
      <FullCalendar
        plugins={[dayGridPlugin]}
        initialView="dayGridMonth"
      />
    </div>
  )
}
