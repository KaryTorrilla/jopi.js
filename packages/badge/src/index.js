import React from 'react'
import { Box } from '@oneloop/box'
import '@oneloop/fonts'

export const Badge = ({ isMedium, hasIcon, isNotButton, ...props }) => (
  <Box
    as={ isNotButton ? "" : "button" }
    tx="buttons"
    variant="primary"
    {...props}
    __css={{
      lineHeight: '14px',
      height: isMedium ? '22px' : '16px',
      display: 'inline-block',
      fontSize: '10px',
      fontFamily: 'primary',
      textAlign: 'center',
      textTransform: 'uppercase',
      fontWeight: 'bold',
      letterSpacing: '0.05em',
      px: 2,
      py: '2px',
      color: 'white',
      bg: 'primary',
      border: 0,
      borderRadius: isMedium ? '14px' : '8px',
      flexDirection: 'row',
      justifyContent: 'center',
      alignItems: 'center',
      padding: hasIcon ? '4px 8px' : '1px 5px',
      position: 'static',
      flex: 'none',
      order: 0,
      flexGrow: 0,
      margin: '0px 0px',
    }}
  />
)
