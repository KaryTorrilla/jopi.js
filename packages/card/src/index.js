import React from 'react'
import { Box } from '@oneloop/box'
import theme from '@oneloop/theme'

export const Card = ({ variant, active, ...props }) => {
  const variantValues = Object.values(theme.card)[Object.keys(theme.card).indexOf(variant)]
  const backgroundColor = active ? variantValues[':focus'].backgroundColor : variantValues.backgroundColorValue
  return (
    <Box sx={{ position: 'relative' }}>
      <Box
        tx='card'
        variant={variant}
        {...props}
        __css={{
          fontFamily: 'primary',
          display: 'flex',
          flexDirection: 'row',
          alignItems: 'center',
          backgroundColor: backgroundColor,
          position: 'static',
        }}
        />
    </Box>
  )
}
