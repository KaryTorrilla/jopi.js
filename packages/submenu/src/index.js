import React from 'react'
import { createPopper } from '@popperjs/core'
import { Box } from '@oneloop/box'

export const Submenu = ({ parentId, childrenId, variant = 'submenu', placement = 'right', ...props }) => {
  const popcorn = document.querySelector('#' + parentId)
  const tooltip = document.querySelector('#' + childrenId)
  if (popcorn != null && tooltip != null) {
    createPopper(popcorn, tooltip, {
      placement: placement,
      modifiers: [
        {
          name: 'offset',
          options: {
            offset: [0, 8],
          },
        },
      ],
    })
  }

  return (
    <Box sx={{ position: 'relative' }}>
      <Box
        tx='panelSubmenu'
        variant={variant}
        {...props}
        __css={{
          appearance: 'none',
          display: 'flex',
          alignItems: 'flex-start',
          whiteSpace: 'nowrap',
          border: 'none',
          fontFamily: 'Nunito Sans',
          background: '#FFFFFF',
          justifyContent: 'center',
        }}
      />
    </Box>
  )
}
