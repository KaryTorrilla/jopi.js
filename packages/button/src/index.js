import React from 'react'
import { Badge } from '@oneloop/badge'
import { Box } from '@oneloop/box'
import { Checkbox } from '@oneloop/checkbox'
import { Icon } from '@oneloop/icons'
import { Image } from '@oneloop/image'
import { useToggle } from '@oneloop/hooks'
import theme from '@oneloop/theme'
import '@oneloop/fonts'

export const Button = ({ variant, ...props }) => (
  <Box sx={{ position: 'relative' }}>
    <Box
      as='button'
      tx='buttons'
      variant={variant}
      {...props}
      __css={{
        appearance: 'none',
        display: 'flex',
        lineHeight: 'inherit',
        fontFamily: 'Nunito Sans',
        fontWeight: 'normal',
        textAlign: 'center',
        textDecoration: 'none',
        cursor: 'pointer',
        outline: 'none',
        color: 'white',
        bg: 'primary',
        border: 0,
        borderRadius: '12px',
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
        whiteSpace: 'nowrap',
      }}
    />
  </Box>
)

export const ButtonIcon = ({ icon, variant, text, badgeValue = 0, badgeVariant = 'primary', maxWidth, ...props }) => {
  let fontSize
  if (Array.isArray(variant)) {
    const indexes = variant.map(v => Object.keys(theme.buttons).indexOf(v))
    indexes.map(index => {
      if (Object.values(theme.buttons)[index].fontSize !== undefined) {
        fontSize = Object.values(theme.buttons)[index].fontSize
      }
      return fontSize
    })
  } else {
    const index = Object.keys(theme.buttons).indexOf(variant)
    fontSize = Object.values(theme.buttons)[index].fontSize
  }

  return (
    <Box sx={ maxWidth ? { position: 'relative', width: '100%' } : { position: 'relative' }}>
      <Box
        as='button'
        tx='buttons'
        variant={variant}
        {...props}
        __css={{
          appearance: 'none',
          display: 'flex',
          lineHeight: 'inherit',
          fontFamily: 'Nunito Sans',
          fontWeight: 'bold',
          textAlign: 'center',
          textDecoration: 'none',
          cursor: 'pointer',
          outline: 'none',
          border: 0,
          flexDirection: 'row',
          justifyContent: 'center',
          alignItems: 'center',
          whiteSpace: 'nowrap',
          position: 'relative',
        }}
      >
        { icon && <Icon icon={icon} fontSize={fontSize}/> }
        { badgeValue !== 0 && <Badge variant={badgeVariant} isNotButton style={{ position: 'absolute', top: '2px', left: '16px' }}>{ badgeValue }</Badge> }
        { text && <span>{text}</span> }
      </Box>
    </Box>
  )
}

export const ButtonHoldPress = ({ variant, active = false, text, icon, badgeValue = 0, badgeVariant = 'primary', isCollapsible, hasCheckbox = false, disabled = false, maxWidth, ...props }) => {
  let colorValue
  let colorValueIcon
  let fontSize
  let fontWeight
  if (Array.isArray(variant)) {
    const indexes = variant.map(v => Object.keys(theme.buttons).indexOf(v))
    indexes.map(index => {
      if (Object.values(theme.buttons)[index][':focus'] !== undefined && active) {
        colorValue = Object.values(theme.buttons)[index][':focus'].color
        fontWeight = Object.values(theme.buttons)[index][':focus'].fontWeight
        if (Object.values(theme.buttons)[index].colorIcon !== undefined) {
          colorValueIcon = Object.values(theme.buttons)[index][':focus'].colorIcon
        }
      } else if (Object.values(theme.buttons)[index].color !== undefined) {
        colorValue = Object.values(theme.buttons)[index].color
      } else if (Object.values(theme.buttons)[index].colorIcon !== undefined) {
        colorValueIcon = Object.values(theme.buttons)[index].colorIcon
      } else if (Object.values(theme.buttons)[index].fontWeight !== undefined) {
        fontWeight = Object.values(theme.buttons)[index].fontWeight
      }
      if (Object.values(theme.buttons)[index].fontSize !== undefined) {
        fontSize = Object.values(theme.buttons)[index].fontSize
      }
      return colorValue
    })
  } else {
    const index = Object.keys(theme.buttons).indexOf(variant)
    colorValue = Object.values(theme.buttons)[index].color
    colorValueIcon = Object.values(theme.buttons)[index].colorIcon
    fontSize = Object.values(theme.buttons)[index].fontSize
    fontWeight = Object.values(theme.buttons)[index].fontWeight
    if (active) {
      colorValue = Object.values(theme.buttons)[index][':focus'].color
      colorValueIcon = Object.values(theme.buttons)[index][':focus'].colorIcon
      fontWeight = Object.values(theme.buttons)[index][':focus'].fontWeight
    }
  }

  return (
    <Box sx={ maxWidth ? { position: 'relative', width: '100%' } : { position: 'relative' }}>
      <Box
        as='button'
        tx='buttons'
        variant={variant}
        {...props}
        __css={{
          appearance: 'none',
          display: 'flex',
          lineHeight: 'inherit',
          fontFamily: 'Nunito Sans',
          fontWeight: {fontWeight},
          textAlign: 'center',
          textDecoration: 'none',
          cursor: 'pointer',
          outline: 'none',
          border: 0,
          flexDirection: 'row',
          //  justifyContent: 'center',
          width: maxWidth &&'100%',
          alignItems: 'center',
          whiteSpace: 'nowrap',
        }}
      >
        { hasCheckbox && active && <Checkbox defaultChecked disabled={disabled}/>}
        { hasCheckbox && !active && <Checkbox defaultChecked={false} disabled={disabled}/>}
        { icon && <Icon icon={icon} fontSize={fontSize} style={{ color: ( colorValueIcon || colorValue ) }}/> }
        { text && <span style={colorValue && { color: colorValue, fontWeight: fontWeight }}>{text}</span>}
        { badgeValue === 0 && isCollapsible && <span className='icon-dropdown' style={{ position: 'absolute', right: '10px', color: '#798B97', fontSize: '14px', transform: active ? 'rotate(-180deg)' : 'rotate(0deg)', paddingTop: '4px' }}/> }
        { badgeValue !== 0 && !text && <Badge variant={badgeVariant} isNotButton style={{ position: 'absolute', top: '2px', left: '16px' }}>{badgeValue}</Badge> }
        { badgeValue !== 0 && text && <Badge variant={badgeVariant} isNotButton style={{ position: 'absolute', right: '16px' }}>{badgeValue}</Badge> }
      </Box>
    </Box>
  )
}

export const ButtonUser = ({ variant, srcImage, text, ...props }) => (
  <Box sx={{ position: 'relative' }}>
    <Box
      as='button'
      tx='buttons'
      variant={variant}
      {...props}
      __css={{
        appearance: 'none',
        display: 'flex',
        lineHeight: 'inherit',
        fontFamily: 'Nunito Sans',
        fontWeight: 'bold',
        fontSize: '14px',
        textAlign: 'center',
        textDecoration: 'none',
        cursor: 'pointer',
        outline: 'none',
        color: 'white',
        bg: 'primary',
        border: 0,
        borderRadius: '12px',
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
        height: '48px',
        whiteSpace: 'nowrap',
        padding: '13px 26px 13px 26px',
        gap: '8px',
      }}
    >
      {srcImage && <Image src={srcImage} variant="avatar"/> }
      {!srcImage && <span className='icon-contacto' style={{ fontSize: '24px', height: '24px' }}/>}
      {text}
    </Box>
  </Box>
)
