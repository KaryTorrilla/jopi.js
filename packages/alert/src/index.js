import React from 'react'
import { Flex } from '@oneloop/box'
import { Check, Exclamation, Times } from '@oneloop/icons'
import { Text } from '@oneloop/text'
import { Button } from '@oneloop/button'

export const Alert = ({ type = 'success', text, ...props }) => (
  <Flex
    variant="alert"
    {...props}
    __css={{
      width: '100%',
      bg: 'white',
      borderRadius: 'circle',
      boxShadow: 'active',
      justifyContent: 'space-between',
      p: 1,
    }}
  >
    {type === 'success' && (
      <Check
        bg="success"
        color="white"
        borderRadius="circle"
        size={40}
        p={3}
        mr={5}
      />
    )}
    {type === 'warning' && (
      <Exclamation
        bg="warning"
        color="white"
        borderRadius="circle"
        size={40}
        p={3}
        mr={5}
      />
    )}
    {type === 'danger' && (
      <Times
        bg="danger"
        color="white"
        borderRadius="circle"
        size={40}
        p={3}
        mr={5}
      />
    )}
    <Text variant="body.0" display="flex" alignItems="center" flexGrow={1}>
      {text}
    </Text>
    <Button
      size={32}
      bg="#F6F2F2"
      p={3}
      display="inline-flex"
      alignSelf="center"
      mr={2}
    >
      <Times color="#716F6F" m="auto" />
    </Button>
  </Flex>
)
