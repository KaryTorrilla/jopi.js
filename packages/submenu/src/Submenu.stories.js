import React from 'react'
import { ButtonHoldPress } from '@oneloop/button'
import { useToggle } from '@oneloop/hooks'
import { Text } from '@oneloop/text'
import { Submenu } from '.'

export default {
  component: Submenu,
  title: 'Submenu',
  argTypes: {
    variant: {
      name: 'variant',
      description: 'Variantes disponibles del submenu: [ \'submenu\', \'tooltip\' ]',
    },
    placement: {
      name: 'placement',
      description: 'PosiciÃ³n del tooltip o submenu: [ \'right\', \'left\', \'up\', \'down\' ]',
      type: 'text',
      control: { type: 'none' },
    },
    parentId: {
      name: 'parentId',
      required: true,
      description: 'Es el Id del componente desde donde saldra el submenu.',
      type: 'text',
    },
    childrenId: {
      name: 'childrenId',
      required: true,
      description: 'Es el Id del componente que encapsula el Submenu.',
      type: 'text',
    },
  },
}

export const SubmenuNormal = () => {
  const [hover, setHover] = useToggle(false)
  const [active, setActive] = useToggle(false)
  return (
    <div style={{ background: '#F3F6F8', padding: '20px', paddingTop: '125px', borderRadius: '10px', height: '135px' }}>
      <ButtonHoldPress variant='smallIconMainButton' icon='icon-propiedades' active={active} badgeVariant='primary' onClick={() => setActive(active)} onMouseOver={() => setHover(true)} onMouseOut={() => setHover(false)} id="parent0" aria-describedby="tooltip0"/>
      <div id="tooltip0" role="tooltip0" hidden={!active && !hover}>
        <Submenu parentId='parent0' childrenId='tooltip0' width='202px'>
          <Text variant='submenuTitle'>Crear</Text>
          <ButtonHoldPress variant='submenu' icon='icon-contacto' text='Contacto' maxWidth/>
          <ButtonHoldPress variant='submenu' icon='icon-empresa' text='Empresa' maxWidth/>
          <ButtonHoldPress variant='submenu' icon='icon-propiedades' text='Propiedad' maxWidth/>
          <ButtonHoldPress variant='submenu' icon='icon-emprendimiento' text='Emprendimiento' maxWidth/>
          <ButtonHoldPress variant='submenu' icon='icon-email' text='Email' maxWidth/>
        </Submenu>
      </div>
    </div>
  )
}

export const SubmenuWithOutIcon = () => {
  const [hover, setHover] = useToggle(false)
  const [active, setActive] = useToggle(false)
  return (
    <div style={{ background: '#F3F6F8', padding: '20px', paddingTop: '125px', borderRadius: '10px', height: '135px' }}>
      <ButtonHoldPress variant='smallIconMainButton' icon='icon-propiedades' active={active} badgeVariant='primary' onClick={() => setActive(active)} onMouseOver={() => setHover(true)} onMouseOut={() => setHover(false)} id="parent1" aria-describedby="tooltip1"/>
      <div id="tooltip1" role="tooltip1" hidden={!active && !hover}>
        <Submenu parentId='parent1' childrenId='tooltip1' width='202px'>
          <Text variant='submenuTitle'>Crear</Text>
          <ButtonHoldPress variant='submenu' text='Contacto' maxWidth/>
          <ButtonHoldPress variant='submenu' text='Empresa' maxWidth/>
          <ButtonHoldPress variant='submenu' text='Propiedad' maxWidth/>
          <ButtonHoldPress variant='submenu' text='Emprendimiento' maxWidth badgeValue={1850} />
          <ButtonHoldPress variant='submenu' text='Email' maxWidth badgeValue={30} />
        </Submenu>
      </div>
    </div>
  )
}

export const Tooltip = () => {
  const [hover, setHover] = useToggle(false)
  const [active, setActive] = useToggle(false)

  return (
    <div style={{ background: '#F3F6F8', padding: '20px', borderRadius: '10px' }}>
      <ButtonHoldPress variant='smallIconMainButton' icon='icon-propiedades' active={active} badgeVariant='primary' onClick={() => setActive(active)} onMouseOver={() => setHover(true)} onMouseOut={() => setHover(false)} id="parent2" aria-describedby="tooltip2"/>
      <div id="tooltip2" role="tooltip2" hidden={!active && !hover}>
        <Submenu parentId='parent2' childrenId='tooltip2' variant='tooltip'>
          <Text variant='body.fontSiz13'>Sitios Webs</Text>
        </Submenu>
      </div>
    </div>
  )
}
