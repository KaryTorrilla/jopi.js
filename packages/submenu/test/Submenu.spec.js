import React from 'react'
import renderer from 'react-test-renderer'
import Enzyme, { mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import 'jest-styled-components'

import { ButtonIcon } from '@oneloop/button'
import { ParentSubmenu, Submenu } from '../src'

Enzyme.configure({ adapter: new Adapter() })

describe('Submenu', () => {
  beforeAll(() => {
    const div = document.createElement('div')
    window.domNode = div
    document.body.appendChild(div)
  })

  test('first case', () => {
    const tree = renderer.create(
      <div style={{ background: '#F3F6F8', padding: '20px', borderRadius: '10px' }}>
        <ParentSubmenu id="parent0" aria-describedby="tooltip0" offset={20}>
          <ButtonIcon variant='mainButtonIcon' icon='icon-propiedades' holdPress badgeVariant='primary'/>
        </ParentSubmenu>
        <div id="tooltip0" role="tooltip0" style={{ width: 'fit-content', visibility: 'visible' }}>
          <Submenu parentId='parent0' childrenId='tooltip0'>
            <ButtonIcon variant='submenu' icon='icon-contactos' text='Contacto' maxWidth/>
            <ButtonIcon variant='submenu' icon='icon-empresa' text='Empresa' maxWidth/>
            <ButtonIcon variant='submenu' icon='icon-propiedades' text='Propiedad' maxWidth/>
            <ButtonIcon variant='submenu' icon='icon-emprendimientos' text='Emprendimiento' maxWidth/>
            <ButtonIcon variant='submenu' icon='icon-email' text='Email' maxWidth/>
          </Submenu>
        </div>
      </div>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('renders correctly', () => {
    const wrapper = mount(
      <div style={{ background: '#F3F6F8', padding: '20px', borderRadius: '10px' }}>
        <ParentSubmenu id="parent1" aria-describedby="tooltip1" offset={20} placement='left' hover>
          <ButtonIcon id="button" variant='mainButtonIcon' icon='icon-propiedades' holdPress badgeVariant='primary'/>
        </ParentSubmenu>
        <div id="tooltip1" role="tooltip1" style={{ width: 'fit-content', visibility: 'visible' }}>
          <Submenu parentId='tooltip1' childrenId='tooltip1' placement='left' arrow>
            <ButtonIcon variant='submenu' icon='icon-contactos' text='Contacto' maxWidth/>
            <ButtonIcon variant='submenu' icon='icon-empresa' text='Empresa' maxWidth/>
            <ButtonIcon variant='submenu' icon='icon-propiedades' text='Propiedad' maxWidth/>
            <ButtonIcon variant='submenu' icon='icon-emprendimientos' text='Emprendimiento' maxWidth/>
            <ButtonIcon variant='submenu' icon='icon-email' text='Email' maxWidth/>
          </Submenu>
        </div>
      </div>
      , {
        attachTo: window.domNode,
      })
    expect(wrapper).toMatchSnapshot()
    const parent1 = document.getElementById('parent1')
    parent1.blur()
    parent1.click()
    const join1 = document.getElementById('joinparent1')
    join1.focus()
    join1.blur()
    const tooltip1 = document.getElementById('tooltip1')
    tooltip1.click()
    const btn = document.getElementById('button')
    btn.click()
  })
})
