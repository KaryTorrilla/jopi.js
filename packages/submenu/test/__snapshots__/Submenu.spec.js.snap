// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Submenu renders correctly 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  position: relative;
}

.c1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  line-height: inherit;
  font-family: Nunito Sans;
  font-size: 24px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  outline: none;
  color: white;
  background-color: primary;
  border: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  white-space: nowrap;
}

.c2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  white-space: nowrap;
  border: none;
  font-family: Nunito Sans;
  background: #FFFFFF;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-family: primary;
}

<div
  style={
    Object {
      "background": "#F3F6F8",
      "borderRadius": "10px",
      "padding": "20px",
    }
  }
>
  <div
    className="c0"
  >
    <button
      aria-describedby="tooltipTest"
      className="c1"
      id="parentTest"
    >
      <span
        className="icon-propiedades"
        style={
          Object {
            "color": "#6D777D",
            "height": "22px",
          }
        }
      />
    </button>
  </div>
  <div
    id="tooltipTest"
    role="tooltipTest"
  >
    <div
      className="c0"
    >
      <div
        className="c2"
      >
        <p
          className="c3"
        >
          Sitios Webs
        </p>
      </div>
    </div>
  </div>
</div>
`;
